src/renderer/contexts/ExecutionContext.tsx

```diff
@@ -220,10 +221,13 @@ export const ExecutionProvider = memo(({ children }: React.PropsWithChildren<{}>
         'node-output-data',
         (data) => {
             if (data) {
-                setOutputDataMap((prev) => new Map([...prev, [data.nodeId, data.data]]));
+                const existingData = outputDataMap.get(data.nodeId);
+                if (!existingData || !isJsonEqual(existingData, data.data)) {
+                    setOutputDataMap((prev) => new Map([...prev, [data.nodeId, data.data]]));
+                }
             }
```

Comment:
To not read stale data (`outputDataMap`), move all of this into `setOutputDataMap`. React won't update a `useState` if the value return by the set-supplier function is that same as the input value.